packages/tss/src/tss.ts

```diff
@@ -0,0 +1,677 @@
+import {
+  decrypt,
+  encrypt,
+  KeyDetails,
+  KeyType,
+  keyTypeToCurve,
+  LocalMetadataTransitions,
+  Point,
+  ReconstructedKeyResult,
+  ShareStore,
+  TKeyArgs,
+} from "@tkey/common-types";
+import ThresholdKey, { CoreError, Metadata } from "@tkey/core";
+import { dotProduct, ecPoint, hexPoint, PointHex, randomSelection, RSSClient } from "@toruslabs/rss-client";
+import BN from "bn.js";
+import { ec as EC } from "elliptic";
+import { keccak256 } from "ethereum-cryptography/keccak";
+
+import { TSSTorusServiceProvider } from ".";
+import { FactorEnc, IAccountSaltStore, InitializeNewTSSKeyResult } from "./common";
+import {
+  BasePoint,
+  generateSalt,
+  getLagrangeCoeffs,
+  getPubKeyPoint,
+  kCombinations,
+  lagrangeInterpolation,
+  pointToElliptic,
+  pointToHex,
+} from "./util";
+
+export const TSS_MODULE = "tssModule";
+
+// This has to be fixed to secp256k1 for now because it is being used in
+// combination with eccrypto's `encrypt` and `decrypt`, which is secp256k1 only.
+export const FACTOR_KEY_TYPE = KeyType.secp256k1;
+
+export interface TSSTKeyArgs extends TKeyArgs {
+  serviceProvider: TSSTorusServiceProvider;
+  tssKeyType: KeyType;
+  tssTag?: string;
+}
+
+export interface TKeyInitArgs {
+  withShare?: ShareStore;
+  importKey?: Buffer;
+  neverInitializeNewKey?: boolean;
+  transitionMetadata?: Metadata;
+  previouslyFetchedCloudMetadata?: Metadata;
+  previousLocalMetadataTransitions?: LocalMetadataTransitions;
+  delete1OutOf1?: boolean;
+}
+
+export interface TKeyTSSInitArgs extends TKeyInitArgs {
+  deviceTSSShare?: BN;
+  deviceTSSIndex?: number;
+  factorPub?: Point;
+}
+
+export class TKeyTSS extends ThresholdKey {
+  serviceProvider: TSSTorusServiceProvider = null;
+
+  private _tssKeyType: KeyType;
+
+  private _tssCurve: EC;
+
+  private _tssTag: string;
+
+  private _accountSalt: string;
+
+  constructor(args?: TSSTKeyArgs) {
+    super(args);
+    const { serviceProvider, storageLayer, tssTag = "default", tssKeyType } = args;
+    this.serviceProvider = serviceProvider;
+    this.storageLayer = storageLayer;
+    this._tssTag = tssTag;
+    this._tssKeyType = tssKeyType;
+    this._tssCurve = new EC(tssKeyType);
+  }
+
+  public get tssTag(): string {
+    return this._tssTag;
+  }
+
+  public get tssKeyType(): KeyType {
+    return this._tssKeyType;
+  }
+
+  public set tssTag(tag: string) {
+    this._tssTag = tag;
+  }
+
+  async initialize(params?: TKeyTSSInitArgs): Promise<KeyDetails> {
```

Comment:
on second thought, should we separate tkey intialize from tss intialize.
we could could have a state/status after tkey initialized indicate that this user/account had 'tag' tss account created.
developer can continue input Factor if existing Tss account or call function below for new tss user

initalizeTss ( or createTssAccount) will have take the input minimal input params

Reply:
Hm, thinking about this, I think it will further complicate things. If you initialize a TKeyTSS instance, then you definitely want a TSS Account, so no point in initializing this assuming you don't want a TSS account. In this case, just use TKey core.

The good thing about how it is currently done is that it is completely consistent with how it is done in `tkey-mpc`, so migration is easy and new bugs should be minimal.

Reply:
Noted that the tssSkip flag is there.
this will allow importTssKey.
Should we add check for existing tssKey instead of just seed in the importTssKey function?

Reply:
yes, similar to how they currently have to decide whether they want to initialize with tss or not. here, it is clear that they want to use tss (because this is tss module), and the default behavior is that a new tss account is created on init. if you dont't want that, you set the skip flag.